core/src/main/java/org/apache/druid/segment/SegmentUtils.java

```diff
@@ -19,22 +19,43 @@
 
 package org.apache.druid.segment;
 
+import com.google.common.hash.HashFunction;
+import com.google.common.hash.Hasher;
+import com.google.common.hash.Hashing;
 import com.google.common.io.Files;
 import com.google.common.primitives.Ints;
 import org.apache.druid.guice.annotations.PublicApi;
 import org.apache.druid.java.util.common.IOE;
+import org.apache.druid.java.util.common.StringUtils;
+import org.apache.druid.timeline.DataSegment;
 
 import java.io.File;
 import java.io.FileInputStream;
 import java.io.IOException;
 import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.Collections;
+import java.util.List;
 
 /**
  * Utility methods useful for implementing deep storage extensions.
  */
 @PublicApi
 public class SegmentUtils
 {
+  private static final HashFunction HASH_FUNCTION = Hashing.sha256();
```

Comment:
Hmm, why is it better?

Reply:
Cryptographically-secure hash functions are typically slower since they need to more work to achieve that property. For example, this perf test observed sha256 to be about an order of magnitude slower than murmur3.